core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -967,6 +935,28 @@ public void visitEntitiesByEntityClass(Solution_ solution, Class<?> entityClass,
         }
     }
 
+    public void visitAllProblemFacts(Solution_ solution, Consumer<Object> visitor) {
+        // Visits facts.
+        for (MemberAccessor accessor : problemFactMemberAccessorMap.values()) {
+            Object object = extractMemberObject(accessor, solution);
+            if (object != null) {
+                visitor.accept(object);
+            }
+        }
+        // Visits problem facts from problem fact collections.
+        for (MemberAccessor accessor : problemFactCollectionMemberAccessorMap.values()) {
+            Collection<Object> objects = extractMemberCollectionOrArray(accessor, solution, true);
+            for (Object object : objects) {
+                visitor.accept(object);
+            }
+        }
+    }
+
+    public void visitAll(Solution_ solution, Consumer<Object> visitor) {
```

Comment:
Technically this changes the original order of iteration.
It was entities by property, facts by property, entities by method, facts by method.
Now it's facts by property, facts by method, entities by property, entities by method.

Does it matter? I'd argue not.